api/src/pcapi/sandboxes/scripts/creators/industrial/create_industrial_invoices.py

```diff
@@ -207,6 +213,144 @@ def create_specific_invoice() -> None:
+            bank_account_id=bank_account.id,
+            cashflow_ids=cashflow_ids,
+        )
+
+        # the default date is used when built, which is now()
+        # but... we would like something more realistic: an invoice
+        # built in the past.
+        invoice = finance_models.Invoice.query.order_by(finance_models.Invoice.id.desc()).first()
+        invoice.date = last_day
+
+        db.session.add(invoice)
+
+        return booking
+
+    # start in the past, move on to today
+    start = datetime.now(timezone.utc) - timedelta(days=15 * count)  # pylint: disable=datetime-now
+
+    beneficiary = None
+    latest_cashflow_batch_id = get_latest_cashflow_batch_id()
+
+    # since invoices and cashflow batches are built in the past,
+    # the batch label generation will not work: the label is
+    # incremented based on the most recent one's.
+    mock_path = "pcapi.core.finance.api._get_next_cashflow_batch_label"
+    with patch(mock_path) as mock_cashflow_label:
+        mock_cashflow_label.side_effect = cashflow_batch_label_generator(latest_cashflow_batch_id, count)
+
+        try:
+            user = users_models.User.query.filter_by(email="activation@example.com").one_or_none()
+            if not user:
+                user = users_factories.ProFactory(
+                    lastName="PRO",
+                    firstName="Activation",
+                    email="activation@example.com",
+                )
+
+            offerer = offerers_factories.OffererFactory(name="Structure avec de nombreux remboursements")
+            offerers_factories.UserOffererFactory(offerer=offerer, user=user)
+
+            current_timestamp = int(datetime.now().timestamp())  # pylint: disable=datetime-now
+            bank_account = finance_factories.BankAccountFactory(
+                label=f"Compte bancaire avec plein de remboursements #{current_timestamp}",
+                offerer=offerer,
+                dsApplicationId=current_timestamp,
+            )
+            venue = offerers_factories.VenueFactory(
+                name="Lieu avec plein de remboursements",
+                managingOfferer=offerer,
+                pricing_point="self",
+                bank_account=bank_account,
+            )
+            offer = offers_factories.OfferFactory(venue=venue, dateCreated=start)
+
+            for idx in range(1, count + 1):
+                booking = book_offer_and_build_invoices(
+                    current_idx=idx,
+                    start=start,
+                    beneficiary=beneficiary,
+                    offer=offer,
+                    bank_account=bank_account,
+                )
+
+                # first booking will create a beneficiary,
+                # no need to create a new one for each booking
+                if not beneficiary and booking:
+                    beneficiary = booking.user
+        except Exception:
+            db.session.rollback()
+            raise
+
+        db.session.commit()
```

Comment:
Est-ce qu'il ne faudrait pas remonter le `db.session.commit()` comme ceci ?

```suggestion
                if not beneficiary and booking:
                    beneficiary = booking.user
                db.session.commit()
        except Exception:
            db.session.rollback()
            raise
```